README.md

```diff
@@ -1,94 +1,158 @@
-* Learn [Angular 2 for Dart](https://angular.io/docs/dart/latest/quickstart.html) - *Only for performance dashboard*
+# Using Cocoon
 
-# Running local dev server
+## Agents
 
-This is useful for developing backend functionality locally. This local dev
-server can be connected to the frontend applications by running `dart dev/deploy.dart --project test --version test`
-and answer `N` to deploying to AppEngine. This will build the frontend files
-and copy them to the directory the server will serve them out of.
+The dashboard has a page showing [the status of all our current
+agents](https://flutter-dashboard.appspot.com/#/agents).
 
-Set the environment variables `GCLOUD_PROJECT` and `GCLOUD_KEY`. Running the
-following command will give more explaination on what these values should be.
+### Creating an agent
 
-[Make sure to create a service account in the GCP dashboard](https://pantheon.corp.google.com/iam-admin/serviceaccounts?project=flutter-dashboard&supportedpurview=project)
+To create an agent in the dashboard, it needs an `agentId` and a list
+of capabilities (comma delimited). Clicking the "Create Agent" button
+will show a dialog for creating an agent.
 
-`cd app_dart && dart bin/server.dart`
+> An example of a valid agent would be `agentId`=`bot-with-devices` and
+> `capabilities`=`has-android-phone,has-iphone`.
 
-If you see `Serving requests at 0.0.0.0:8080` the dev server is working.
+The dialog returns returns an authentication token, and prints it to
+the console. This token is not stored, so copy it immediately and add
+it to the agent's configuration file. If the token is lost or
+compromised, authorize the agent to generate a new token.
 
-# Building Cocoon for deployment
+### Authorizing an agent
 
-The following command will run tests and build the app, and provide instructions
-for deploying to Google App Engine.
+Click on the dropdown for the agent, and click authorize agent. This
+will print a new generated token to the console.
 
-```sh
-cd app_dart
-dart dev/deploy.dart --project {PROJECT} --version {VERSION}
-```
+This command invalidates any previously issued authentication tokens
+for the given agent. Only one authentication token is valid at any
+given moment in time. Therefore, if the agent is currently using a
+previously issued token its API requests will be rejected until it
+switches to using the newly created token.
 
-You can test the new version by accessing `{VERSION}-dot-flutter-dashboard.appspot.com` in your
-browser. If the result is satisfactory, the new version can be activated by using the Cloud Console
-UI: https://pantheon.corp.google.com/appengine/versions?project=flutter-dashboard&serviceId=default
+## Forcing a refresh from GitHub
 
-## Optional flags
+The server is driven by commits made to
+https://github.com/flutter/flutter repo. It periodically syncs new
+commits. If you need to manually force a refresh, query
+`https://flutter-dashboard.appspot.com/api/refresh-github-commits`.
```

Comment:
This is only allowed for users that have auth access to Cocoon (Googlers and whitelisted accounts)